### PR TITLE
Bmff v2

### PR DESCRIPTION
**Pull request by mauricefisher64**
## Changes in this pull request
Support BMFF V2 hashing and validation.  BMFF V2 requires hashing of the box location before hashing the box contents.  This required hasher changes to be able to pause and insert the hashes at the correct position.

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [x] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/store.rs [hunk 1]
```diff
@@ -1337,12 +1337,15 @@ impl Store {
 
     // generate a list of AssetHashes based on the location of objects in the stream
     #[cfg(feature = "sign")]
-    fn generate_data_hashes_for_stream(
-        stream: &mut dyn CAIReadWrite,
+    fn generate_data_hashes_for_stream<R>(
+        stream: &mut R,
```
Comment (scouten-adobe) [inline sdk/src/store.rs:1341]: This seems like a good change. :+1:

Comment (mauricefisher64) [reply sdk/src/store.rs:1341]: Yeah, now we can accept anything that has a Read + Seek trait.

**Review by scouten-adobe — APPROVED** [review]

File: sdk/src/assertions/bmff_hash.rs [hunk 2]
```diff
@@ -255,13 +270,16 @@ impl AssertionCbor for BmffHash {}
 
 impl AssertionBase for BmffHash {
     const LABEL: &'static str = Self::LABEL;
-    const VERSION: Option<usize> = Some(ASSERTION_CREATION_VERSION);
+    const VERSION: Option<usize> = Some(ASSERTION_CREATION_VERSION); // todo: this mechanism needs to change since a struct could support different versions
```
Comment (gpeacock) [inline sdk/src/assertions/bmff_hash.rs:273]: Would you ever be generating different versions of the assertion in the same claim?

**Review by gpeacock — APPROVED** [review]
Looks good, one minor comment and a question.

File: sdk/src/assertions/bmff_hash.rs (hunk 2 above)
Comment (mauricefisher64) [reply sdk/src/assertions/bmff_hash.rs:273]: No we always generate the latest supported.

**Pull request merged**